flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/MaxAggregate.scala

```diff
@@ -17,68 +17,96 @@
  */
 package org.apache.flink.api.table.runtime.aggregate
 
+import org.apache.calcite.sql.`type`.SqlTypeName
+import org.apache.flink.api.table.Row
+
 abstract class MaxAggregate[T: Numeric] extends Aggregate[T] {
 
-  var result: T = _
-  val numericResult = implicitly[Numeric[T]]
+  private val numeric = implicitly[Numeric[T]]
 
-  override def aggregate(value: Any): Unit = {
-    val input: T = value.asInstanceOf[T]
+  /**
+   * Initiate the partial aggregate value in Row.
+   * @param intermediate
+   */
+  override def initiate(intermediate: Row): Unit = {
+    intermediate.setField(aggOffsetInRow, numeric.zero)
```

Comment:
should initialize to the minimum value of the data type instead of `numeric.zero`.